File: Application.Domain/Model/Education/Dtos/CourseImportResult.cs
namespace Application.Domain.Model.Education.Dtos
{
    public class CourseImportResult
    {
        public int Processed { get; set; }
        public int CreatedSchools { get; set; }
        public int CreatedPrograms { get; set; }
        public int CreatedClasses { get; set; }
        public int CreatedUcs { get; set; }
        public int UpdatedUcs { get; set; }
        public int Linked { get; set; }
        public List<CourseImportError> Errors { get; } = new();
    }

    public class CourseImportError
    {
        public string? CourseCategory { get; set; }
        public string? UcName { get; set; }
        public string Message { get; set; } = string.Empty;
    }
}

File: Application.Client/src/app/model/course-import-result.ts
export interface CourseImportResult {
  Processed: number;
  CreatedSchools: number;
  CreatedPrograms: number;
  CreatedClasses: number;
  CreatedUcs: number;
  UpdatedUcs: number;
  Linked: number;
  Errors: CourseImportError[];
}

export interface CourseImportError {
  CourseCategory?: string;
  UcName?: string;
  Message: string;
}
